rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -101,6 +105,14 @@
     @Context
     private UriInfo uriInfo;
 
+    /**
+     * Only lower-case letters and digits; min 10 chars; max 1024 chars.
+     * 
+     * We are this extreme because some existing entity implementations rely on the entity-id format 
+     * for use in hostnames, etc.
+     */
+    private final Pattern appIdPattern = Pattern.compile("[a-z0-9]{10,1022}");
```

Comment:
For max length, I don't know what a good number is! If we're worried about it being used in cloud resource names, we might need to restrict it to < 64 chars. but that seems far too restrictive for ids (coming back to Mark's point about the use of proper UUID generators).